### ISSUE TITLE
Incorrect href link for ONE_TO_ONE relation with OID as id type

### ISSUE THREAD

**Issue by shaileshpandit** [SoftInstigate/restheart#279]
After creating a link to other document with OID as id type, the href in link is incorrect

## Expected Behavior
href in owning document should be object id string instead of whole json object
"href": "/test/collother/5ae94eb4b634b2c96dc9f242"

## Current Behavior
href in owning document is whole json object
 "href": "/test/collother/{\"$oid\":\"5ae94eb4b634b2c96dc9f242\"}"

## Context
was trying to simply link other document from one document

## Environment
05:19:20.724 [main] INFO  org.restheart.Bootstrapper - version 3.2.2
05:19:21.687 [main] INFO  org.restheart.Bootstrapper - MongoDB version 3.6.4,

## Steps to Reproduce
```
http -a a:a PUT localhost:8080/test
	HTTP/1.1 201 Created
```

```
http -a a:a PUT localhost:8080/test/collother
	HTTP/1.1 201 Created
```

```
http -a a:a POST localhost:8080/test/collother name="test"
	HTTP/1.1 201 Created
	Location: http://localhost:8080/test/collother/5ae94eb4b634b2c96dc9f242
```

```
http -a a:a PUT localhost:8080/test/collowing rels:='[{"rel":"other","type":"ONE_TO_ONE","role":"OWNING","target-coll":"collother","ref-field":"otherId"}]'
	HTTP/1.1 201 Created
```

```
http -a a:a POST localhost:8080/test/collowing addr="tetsaddr" otherId:='{"$oid":"5ae94eb4b634b2c96dc9f242"}'
	HTTP/1.1 201 Created
	Location: http://localhost:8080/test/collowing/5ae95182b634b2c96dc9f2c1
```

```
http -a a:a GET localhost:8080/test/collowing/5ae95182b634b2c96dc9f2c1
	HTTP/1.1 200 OK
	{
	    "_etag": {
	        "$oid": "5ae9518263445900055d3984"
	    },
	    "_id": {
	        "$oid": "5ae95182b634b2c96dc9f2c1"
	    },
	    "_links": {
	        "other": {
	            "href": "/test/collother/{\"$oid\":\"5ae94eb4b634b2c96dc9f242\"}"
	        }
	    },
	    "addr": "tetsaddr",
	    "otherId": {
	        "$oid": "5ae94eb4b634b2c96dc9f242"
	    }
	}
```

```
http -a a:a GET localhost:8080/test/collother/{\"\$oid\":\"5ae94eb4b634b2c96dc9f242\"}
	HTTP/1.1 404 Not Found
	{
	    "http status code": 404,
	    "http status description": "Not Found",
	    "message": "document '{'$oid':'5ae94eb4b634b2c96dc9f242'}' does not exist"
	}
```

```
http -a a:a GET localhost:8080/test/collother/5ae94eb4b634b2c96dc9f242
	HTTP/1.1 200 OK
	{
	    "_etag": {
	        "$oid": "5ae94eb463445900055d397f"
	    },
	    "_id": {
	        "$oid": "5ae94eb4b634b2c96dc9f242"
	    },
	    "name": "test"
	}
```

**Comment by mkjsix** [comment]
@ujibang this looks like a regression, I've never noticed such an issue before.

**Comment by ujibang** [comment]
fixed in 6c397e741276766eaae8f9bab4873c52543144af

**Comment by mkjsix** [comment]
We're going to release version 3.3 final this week, it will include this fix as well.

**Comment by mkjsix** [comment]
@shaileshpandit yesterday we've released the fix in both RESTHeart 3.3.0 and 3.2.3. Beware that version 3.3 introduces some different behaviors for PUT and POST, so have a look at the release notes before jumping: https://github.com/SoftInstigate/restheart/releases

**Comment by shaileshpandit** [comment]
Thanks, works as expected in 3.3.0
